runtime/dom-context.js

```diff
@@ -26,6 +26,12 @@ class DomContext {
     this._liveDom = null;
     this._innerContextBySlotName = {};
   }
+  static createContext(context, content) {
+    let domContext = new DomContext(context);
+    domContext.stampTemplate(DomContext.createTemplateElement(content.template), () => {});
```

Comment:
Fwiw, there is optimization to be had by making only one template for each unique string (Xen memoizes information parsed from the template the first time it's used.) Mapping against the String reference might work, I'm not sure exactly what the memory pattern looks like for Strings sent over the message bus. This is not all that important for now, but it's a useful long-term goal to avoid duplicating Template objects.

Reply:
yeah, we already store the templates in dom-slot (for the particle templates). I'll change that to be a more generic solution in a separate PR.